src/hotspot/share/gc/parallel/parallelScavengeHeap.inline.hpp

```diff
@@ -40,6 +40,13 @@ inline bool ParallelScavengeHeap::should_alloc_in_eden(const size_t size) const
 
 inline void ParallelScavengeHeap::invoke_scavenge() {
   PSScavenge::invoke();
+  capture_live();
+}
+
+inline void ParallelScavengeHeap::capture_live() {
+  _young_live = young_gen()->used_in_bytes();
+  _eden_live = young_gen()->eden_space()->used_in_bytes();
+  _old_live = old_gen()->used_in_bytes();
```

Comment:
`_young_live` already seems to contain `_eden_live` looking at the implementation of `PSYoungGen::used_in_bytes()`:

I.e.

`size_t PSYoungGen::used_in_bytes() const {
  return eden_space()->used_in_bytes()
       + from_space()->used_in_bytes();      // to_space() is only used during scavenge
}
`

but maybe I'm wrong here.

Reply:
This seems like a correct summary - I have updated the implementation.